server/routes/user.ts

```diff
@@ -108,6 +108,28 @@ router.put<{ id: string }>('/:id', async (req, res, next) => {
       });
     }
 
+    // To protect against sql injections via api
+    if (!req.body.username.match(/^[a-zA-Z0-9]+$/)) {
+      return next({
+        status: 500,
+        message: 'Invalid username format',
+      });
+    }
```

Comment:
Using the query builder will prevent SQL injection.

Reply:
I don't think we need to limit usernames so severely. Maybe a max character limit? We should allow spaces/dashes/other special characters if someone wants them.

Reply:
Alright, I didn't know that. I think we should still limit it so that it doesn't destroy formatting in notifications if they use html/markdown characters in their name. Maybe `[a-zA-Z0-9\-\_]`?

Reply:
Spaces as well? `\s`